test/integration/scheduler/predicates_test.go

```diff
@@ -1036,3 +1037,54 @@ var (
 	hardSpread = v1.DoNotSchedule
 	softSpread = v1.ScheduleAnyway
 )
+
+func TestUnschedulablePodBecomesSchedulable(t *testing.T) {
+	tests := []struct {
+		name   string
+		init   func(kubernetes.Interface) error
+		pod    *pausePodConfig
+		update func(kubernetes.Interface) error
+	}{
+		{
+			name: "node gets added",
+			pod: &pausePodConfig{
+				Name: "pod-1",
+			},
+			update: func(cs kubernetes.Interface) error {
+				_, err := createNode(cs, "node-1", nil)
+				return err
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
```

Comment:
I didn't see this pattern is used in integration test before. So I suppose it won't work as expected?

> // Parallel signals that this test is to be run in parallel with (and only with)
> // other parallel tests.

Reply:
yup, creating nodes/pods with the same name won't make etcd happy.